pkg/registrymirror/containerd/utils.go

```diff
@@ -0,0 +1,39 @@
+package containerd
+
+import (
+	"net/url"
+	"strings"
+)
+
+// ToAPIEndpoint turns URL to a valid API endpoint used in
+// a containerd config file for a local registry.
+func ToAPIEndpoint(URL string) string {
+	cnt := strings.Count(URL, "/")
+	index := -1
+	if cnt > 1 {
+		u, _ := url.Parse(URL)
+		if u.Scheme == "" {
+			u, _ = url.Parse("oci://" + URL)
+			u.Scheme = ""
+		}
+		if u.Path != "" {
+			index = strings.Index(URL, u.Path)
+		}
+	} else {
+		index = strings.Index(URL, "/")
+	}
+	if index == -1 {
+		return URL
+	}
+	return URL[:index] + "/v2" + URL[index:]
+}
```

Comment:
Maybe we could do something different to avoid relying on index based string manipulation, which in my experience can lead to bugs, specially with URLs due to their complexity
For example

```suggestion
func ToAPIEndpoint(url string) string {
	u, err := parseURL(url)
	if err != nil {
		return url
	}

	if u.Path != "" {
		u.Path = filepath.Join("v2", u.Path)
	}

	return strings.TrimPrefix(u.String(), "//")
}

func parseURL(in string) (*url.URL, error) {
	urlIn := in
	if !strings.Contains(in, "//") {
		urlIn = "//" + in
	}

	return url.Parse(urlIn)
}
```

We should also document what happens when the provided url is malformed (it returns the original input?)